packages/@expo/cli/src/start/server/middleware/DevToolsPluginMiddleware.ts

```diff
@@ -0,0 +1,88 @@
+import assert from 'assert';
+import { readFile } from 'fs/promises';
+import path from 'path';
+
+import { ExpoMiddleware } from './ExpoMiddleware';
+import { ServerRequest, ServerResponse } from './server.types';
+import DevToolsPluginManager, { DevToolsPluginEndpoint } from '../DevToolsPluginManager';
+
+export { DevToolsPluginEndpoint };
+
+export class DevToolsPluginMiddleware extends ExpoMiddleware {
+  constructor(
+    projectRoot: string,
+    private readonly pluginManager: DevToolsPluginManager
+  ) {
+    super(projectRoot, [DevToolsPluginEndpoint]);
+  }
+
+  override shouldHandleRequest(req: ServerRequest): boolean {
+    if (!req.url?.startsWith(DevToolsPluginEndpoint)) {
+      return false;
+    }
+    return true;
+  }
+
+  async handleRequestAsync(req: ServerRequest, res: ServerResponse): Promise<void> {
+    assert(req.headers.host, 'Request headers must include host');
+    const { pathname } = new URL(req.url ?? '/', `http://${req.headers.host}`);
+    const pluginName = pathname.substring(DevToolsPluginEndpoint.length + 1).split('/')[0];
+    if (!pluginName) {
+      res.statusCode = 404;
+      res.end();
+      return;
+    }
+    const webpageRoot = await this.pluginManager.queryPluginWebpageRootAsync(pluginName);
+    if (!webpageRoot) {
+      res.statusCode = 404;
+      res.end();
+      return;
+    }
+
+    const staticFilePath = this.shouldServeStaticFile(pluginName, pathname);
+    if (staticFilePath) {
+      const content = (await readFile(path.join(webpageRoot, staticFilePath))).toString('utf-8');
```

Comment:
Maybe we should use something like `import send from 'send';` which we already use for the `public/` folder. This would also have a more robust system for serving mimetypes and streaming results.

Reply:
that's a super neat suggestion. not only having robust implementation but also performant solution because pipe streaming from `send`. good to know we have `send` in cli already.
pr updated, please help to review when you get another chance.